### ISSUE TITLE
Right Hand Rule and Polygons

### ISSUE THREAD

**Issue by ccurran01** [openlayers/openlayers#8861]
I am running into some issues with the right hand rule and polygons.

I am getting errors from knex-postgis when trying to create a geometry from geoJSON ([geomFromGeoJSON](https://postgis.net/docs/ST_GeomFromGeoJSON.html)), claiming that the geoJSON is invalid as it does not follow the right hand rule. 

knex-postgis uses [geoJSON hint](https://github.com/mapbox/geojsonhint) to validate the geojson before attempting to post it to the db. 

When getting the feature as geoJSON from the map we  call `writeFeatureObject` on `GeoJSON` and pass the options
 
```
{
        featureProjection: 'EPSG:27700',
        dataProjection: 'EPSG:27700',
        rightHanded: true
}
```

My understanding of this is that we should get a feature, with co-ordinates in BNG, that follow the right hand rule. 

I'm not sure where I'm going wrong here, have I misunderstood what the flag does? 
Does the coordinate system have some sort of impact on the validity of the geoJSON? 

Been looking at this for a few days now and my head is starting to melt, any pointers would be really useful. 

We are on Open Layers Version 5.1.3

**Comment by geekdenz** [comment]
I believe this should be re-opened for the following reasons:

OpenLayers 6.3.1 still seems to somewhat suffer from this issue.

I have been using both:
`proj4.defs('EPSG:2193', '+proj=tmerc +lat_0=0 +lon_0=173 +k=0.9996 +x_0=1600000 +y_0=10000000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs +axis=neu');`

and

`proj4.defs('EPSG:2193', '+proj=tmerc +lat_0=0 +lon_0=173 +k=0.9996 +x_0=1600000 +y_0=10000000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');`

Note the missing ' +axis=neu' in the second version.

The right-handedness of GeoJSON is somewhat broken in our applications and we are using OpenLayers heavily. If I use the EPSG:2193 projection with ' +axis=neu' the GeoJSON produced by writeFeatureObject is invalid. However without it it is indeed valid.

This makes sense and is related to
https://github.com/openlayers/openlayers/issues/11134 

See this line:
https://github.com/openlayers/openlayers/blob/1ef103ed6/src/ol/source/WMTS.js#L475 on a hint how to fix it.

I'll see if I can confirm this is indeed the case in main (former master) and I can find time to fix this and in that case submit a PR.

**Comment by ahocevar** [comment]
This is probably the same or at least a similar issue as #8148.

**Comment by geekdenz** [comment]
Right now it appears I was mistaken. Although now I am puzzled how I got here...

I've just converted from what I now call EPSG:2193:neu to WGS84 features and back again to GeoJSON using WGS84. It appears the coordinates are correct in 6.3.1. Maybe it had already been fixed or I just happened to have a valid specimen of GeoJSON in 2193.

If someone gets tripped up by this. For future reference, this link helped me shed light on it:
https://www.mathopenref.com/coordpolygonarea2.html#:~:text=The%20parts%20of%20the%20polygon,not%20what%20you%20would%20expect. 

I need to fix an issue related to this in a couple of days so stay tuned.
